Extract commit ID with ttp instead of shell and awk

reihl opened the pull request:
Now in case the command failed the error appear immediately, also the output is more easy to read and understand

fredericlepied commented:
retest

tonyskapunk commented on roles/installer/tasks/23_rhcos_image_paths.yml:
```diff
@@ -2,15 +2,30 @@
     - name: Get COMMIT_ID
-      shell: |
-        /usr/local/bin/{{ installer_cmd }} version | grep '^built from commit' | awk '{print $4}'
-      register: commit_id
```
Could fail sooner:
```yaml
    - name: Get COMMIT_ID
      shell: |
        set -xeo pipefail
        /usr/local/bin/{{ installer_cmd }} version | grep '^built from commit' | awk '{print $4}'
      register: commit_id
    
```

reihl replied:
I agree that this approach is better now, I removed the ttp requirement, however, the test didn't visit the tasks we changed so, I can't tell if the requirement in ansible.util take an action. Can I suggest another way to parse the output? I think we can still use command without pipe and get the information we want with some ansible filters

tonyskapunk replied:
> he test didn't visit the tasks we changed

Correct, it is because this is only applicable for OCP 4.8 or older.  Are you consuming OCP 4.8 or anything older?

> Can I suggest another way to parse the output? I think we can still use command without pipe

Sure

tonyskapunk reviewed (commented):
I agree that is best to avoid shell whenever is possible, but I have some thoughts for this change.

- This is a change for a deprecated version of OCP
- We are adding a dependency (if there is a package for ttp)
- The change removes the shell but ends up executing the command to parse it later

I added a suggestion to fail it sooner, if that would make the experience better, wdyt?

ramperher reviewed (commented):
Can we test this with OCP 4.7? Because this is the target version for this change.

reihl commented:
> Can we test this with OCP 4.7? Because this is the target version for this change.

I can run it on my ci machine, is it enough?

fredericlepied commented:
retest

manurodriguez commented:
> TestDallas: ocp-4.17-vcp

We probably mean `ocp-4.7-vcp`?

tonyskapunk commented:
> > TestDallas: ocp-4.17-vcp
> 
> We probably mean `ocp-4.7-vcp`?

I talked to @reihl and we will be testing with 4.8, I've made the change in the body of the PR

---

recheck

tonyskapunk commented:
> Can we test this with OCP 4.7? Because this is the target version for this change.

Good point, checking this one with @reihl

tonyskapunk commented:
The last failure is due to a maintenance going on in RDO project, I'll retry once it's done.

ramperher commented:
> > > TestDallas: ocp-4.17-vcp
> > 
> > 
> > We probably mean `ocp-4.7-vcp`?
> 
> I talked to @reihl and we will be testing with 4.8, I've made the change in the body of the PR

But the change only affects to OCP 4.7, doesn't it? According to this when condition:

```
  when: (release_version.split('.')[0]|int == 4) and (release_version.split('.')[1]|int <= 7)
```

reihl commented:
> > > > TestDallas: ocp-4.17-vcp
> > > 
> > > 
> > > We probably mean `ocp-4.7-vcp`?
> > 
> > 
> > I talked to @reihl and we will be testing with 4.8, I've made the change in the body of the PR
> 
> But the change only affects to OCP 4.7, doesn't it? According to this when condition:
> 
> ```
>   when: (release_version.split('.')[0]|int == 4) and (release_version.split('.')[1]|int <= 7)
> ```

Correct, I also change the when condition to be more readable with ansible.builtin.version filter. We face it because something that ignore we don't need rhcos.json for version newer then 4.7 and this is the reason I fix it.

tonyskapunk commented:
> But the change only affects to OCP 4.7, doesn't it? According to this when condition:

@ramperher correct, I pinged @reihl to see if he can share logs as he is using OCP 4.8, but the change in here shouldn't affect 4.8, so I'm not sure how exactly this is going to fix something in 4.8

tonyskapunk commented:
>  We face it because something that ignore we don't need rhcos.json for version newer then 4.7 and this is the reason I fix it.

Could you please elaborate a bit more here?

For 4.8 and above a different set of tasks are performed to use `oc  coreos print-stream-json`, if you're using 4.8 this is the tasks that should get you what's needed without having to go to a commit as in 4.7-.

https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/2f66e0f84b3037236cb408aafd57fb8d79b35f9e/roles/installer/tasks/23_rhcos_image_paths.yml#L39-L43

reihl commented:
retest
